pkg/scalers/redis_scaler.go

```diff
@@ -193,110 +190,47 @@ func createRedisScalerWithClient(client *redis.Client, meta *redisMetadata, scri
 	}
 }
 
-func parseTLSConfigIntoConnectionInfo(config *scalersconfig.ScalerConfig, connInfo *redisConnectionInfo) error {
-	enableTLS := defaultEnableTLS
+func parseTLSConfigIntoConnectionInfo(config *scalersconfig.ScalerConfig) (*scalersconfig.ScalerConfig, error) {
```

Comment:
Do you think its a good idea to put all the logic of this method into the `Validate()` method ? It will be similar to the apache kafka scaler where we have complicated value parsing dependency kind of cases https://github.com/kedacore/keda/blob/66d4c9595e917a86b390bbed6eed88b1b7cc5c92/pkg/scalers/apache_kafka_scaler.go#L90-L124